falco.yaml

```diff
@@ -152,525 +535,474 @@ syscall_event_drops:
-# and are exported at a constant stats interval.
-#
-# `libbpf_stats_enabled`:
-# Exposes statistics similar to `bpftool prog show`, providing information such as the number
-# of invocations of each BPF program attached by Falco and the time spent in each program
-# measured in nanoseconds.
-# To enable this feature, the kernel must be >= 5.1, and the kernel configuration `/proc/sys/kernel/bpf_stats_enabled`
-# must be set. This option, or an equivalent statistics feature, is not available for non `*bpf*` drivers.
-# Additionally, please be aware that the current implementation of `libbpf` does not
-# support granularity of statistics at the bpf tail call level.
+# - `0` or `MAX_NUMBER_ONLINE_CPUs` mean a syscall buffer shared between all
+#   CPUs, so 1 buffer
 #
-# todo: prometheus export option
-# todo: syscall_counters_enabled option
+#          CPUs     0  X  2  3  X  X  6  7  8  9   (X means offline CPU)
+#                   |     |  |        |  |  |  |
+#       BUFFERs     0     0  0        0  0  0  0
+#
+# Moreover, you have the option to combine this parameter with
+# `syscall_buf_size_preset` index. For instance, you can create a large shared
+# syscall buffer of 512 MB (using `syscall_buf_size_preset`=10) that is
+# allocated among all the online CPUs.
+#
+# --- [Suggestions]
+#
+# The default choice of index 2 (one syscall buffer for each CPU pair) was made
+# because the modern bpf probe utilizes a different memory allocation strategy
+# compared to the other two drivers (bpf and kernel module). However, you have
+# the flexibility to experiment and find the optimal configuration for your
+# system.
+# 
+# When considering a fixed syscall_buf_size_preset and a fixed buffer dimension:
+# - Increasing this configs value results in lower number of buffers and you can
+#   speed up your system and reduce memory usage
+# - However, using too few buffers may increase contention in the kernel,
+#   leading to a slowdown.
+# 
+# If you have low event throughputs and minimal drops, reducing the number of
+# buffers (higher `cpus_for_each_syscall_buffer`) can lower the memory footprint.
 
-metrics:
-  enabled: false
-  interval: 1h
-  output_rule: true
-  # output_file: /tmp/falco_stats.jsonl
-  resource_utilization_enabled: true
-  kernel_event_counters_enabled: true
-  libbpf_stats_enabled: true
-  convert_memory_to_mb: true
+modern_bpf:
+  cpus_for_each_syscall_buffer: 2
+
+
+#################################################
+# Falco cloud orchestration systems integration #
+#################################################
+
+# [Stable] `metadata_download`
+# These parameters are currently specific to the `k8saudit` plugin.
+# 
+# When connected to an orchestrator like Kubernetes, Falco provides parameters
+# to configure the download behavior. Please note that support for Mesos is
+# deprecated. The following parameters can be used to customize the download
+# behavior:
```

Comment:
```suggestion
# [Stable] `metadata_download`
When connected to an orchestrator, such as Kubernetes, Falco collects metadata, 
enhancing system call events with contextual data.
These parameters control the downloading process of this metadata.

Please note that support for Mesos is deprecated. So these currently apply only 
to the Kubernetes support when enabled using `-k` / `-K` command line flags.

The following parameters can be used to customize the download behavior:
```

These parameters are NOT related to the `k8saudit` plugin. They are actually sinsp settings that take effect when sinsp fetches k8s. for the metadata from the K8S API server to enrich syscall events. This feature is enabled when using `-K/-k` in Falco.
(in the past, those parameters applied to the sinsp Mesos client too, but now it's gone, so they only affect the sinsp K8S client).

Thus, `metadata_download` is a group of knobs that affects the ability to fetch metadata from the k8s API server, influencing Falco's userspace performance when processing syscalls.

So I've tried to correct the description. I just focused on the content, so feel free to change the style or adapt it as you prefer.